TX.h

```diff
@@ -538,15 +538,18 @@ static inline void processSBUS(uint8_t c)
     if ((frameIndex == 23) && (c == SBUS_TAIL)) {
       uint8_t set;
       for (set = 0; set < 2; set++) {
-        PPM[(set<<3)] = ppmWork.sbus.ch[set].ch0 >> 1;
-        PPM[(set<<3)+1] = ppmWork.sbus.ch[set].ch1 >> 1;
-        PPM[(set<<3)+2] = ppmWork.sbus.ch[set].ch2 >> 1;
-        PPM[(set<<3)+3] = ppmWork.sbus.ch[set].ch3 >> 1;
-        PPM[(set<<3)+4] = ppmWork.sbus.ch[set].ch4 >> 1;
-        PPM[(set<<3)+5] = ppmWork.sbus.ch[set].ch5 >> 1;
-        PPM[(set<<3)+6] = ppmWork.sbus.ch[set].ch6 >> 1;
-        PPM[(set<<3)+7] = ppmWork.sbus.ch[set].ch7 >> 1;
+        PPM[(set<<3)] = (((ppmWork.sbus.ch[set].ch0 >> 1) * 5) >> 2);
+        PPM[(set<<3)+1] = (((ppmWork.sbus.ch[set].ch1 >> 1) * 5) >> 2);
+        PPM[(set<<3)+2] = (((ppmWork.sbus.ch[set].ch2 >> 1) * 5) >> 2);
+        PPM[(set<<3)+3] = (((ppmWork.sbus.ch[set].ch3 >> 1) * 5) >> 2);
+        PPM[(set<<3)+4] = (((ppmWork.sbus.ch[set].ch4 >> 1) * 5) >> 2);
+        PPM[(set<<3)+5] = (((ppmWork.sbus.ch[set].ch5 >> 1) * 5) >> 2);
+        PPM[(set<<3)+6] = (((ppmWork.sbus.ch[set].ch6 >> 1) * 5) >> 2);
+        PPM[(set<<3)+7] = (((ppmWork.sbus.ch[set].ch7 >> 1) * 5) >> 2);
       }
+      for(uint8_t i=0; i<16; i++){
+       PPM[i]=(PPM[i]<=108)?0:(PPM[i]-108);
```

Comment:
@mikeller,
yeah, I realize that, but that's  just the way @kh4 designed it, in order to provide full range while keeping the data within the 10-bit range. The same thing happens with PPM input; if your taranis feeds the OLRSng TX a PPM value that's less than 1000us or greater than 2000us, it'll get scaled differently & will have larger resolution steps. This way, @kh4 can still provide full stick range (>=1000us) while keeping the data within 10 bits.

This new math simply uses a linear formula to scale the raw SBUS values into the corresponding raw PPM values. It uses the endpoints & center point to build the formula. The reason for this is just due to the weird scaling that SBUS does. (FYI, this is the same basic formula that is used in baseflight in order to correct the SBUS values)

feel free to test it out if you can, and let me know if it doesn't work as expected.